aws/internal/service/directconnect/finder/finder.go

```diff
@@ -0,0 +1,155 @@
+package finder
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// ConnectionByID returns the connections corresponding to the specified connection ID.
+// Returns NotFoundError if no connection is found.
+func ConnectionByID(conn *directconnect.DirectConnect, connectionID string) (*directconnect.Connection, error) {
+	input := &directconnect.DescribeConnectionsInput{
+		ConnectionId: aws.String(connectionID),
+	}
+
+	connections, err := Connections(conn, input)
+
+	if tfawserr.ErrMessageContains(err, directconnect.ErrCodeClientException, "Could not find Connection with ID") {
+		return nil, &resource.NotFoundError{
+			LastError:   err,
+			LastRequest: input,
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Handle any empty result.
+	if len(connections) == 0 {
+		return nil, &resource.NotFoundError{
+			Message:     "Empty result",
+			LastRequest: input,
+		}
+	}
+
+	if state := aws.StringValue(connections[0].ConnectionState); state == directconnect.ConnectionStateDeleted {
+		return nil, &resource.NotFoundError{
+			Message:     state,
+			LastRequest: input,
+		}
+	}
+
+	return connections[0], nil
+}
+
+// Connections returns the connections corresponding to the specified input.
+// Returns an empty slice if no connections are found.
+func Connections(conn *directconnect.DirectConnect, input *directconnect.DescribeConnectionsInput) ([]*directconnect.Connection, error) {
```

Comment:
I think the naming could be more verbose here since `Connections` can apply to many types of resources.

Reply:
Let's hold off on merging this one until the whole `finder`, `lister` package thing is decided on (#15945 et al.).

Reply:
:+1: